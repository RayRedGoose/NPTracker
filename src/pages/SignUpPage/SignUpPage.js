import './SignUpPage.scss'
import React from 'react'
import SignUpBlock from 'containers/SignUpBlock/SignUpBlock'

const SignUpPage = () => {
  const blockInfo = [['username', 'password'], ['name', 'last_name', 'city'], ['image'], []]
  const blocks = blockInfo.map((info, ind) => {
    return <SignUpBlock key={ind+1} id={ind+1} inputs={info} />
  })

  return (
    <main className="signup-page">
      { blocks }
    </main>
  )
}

export default SignUpPage
