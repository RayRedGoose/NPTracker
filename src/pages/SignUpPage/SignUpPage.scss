@import "../../variables.scss";

.signup-page {
  padding: 0;
  display: flex;
  section {
    width: 25%; height: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}
