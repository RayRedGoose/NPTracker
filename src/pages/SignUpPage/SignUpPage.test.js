import React from 'react';
import { shallow } from 'enzyme';
import SignUpPage from './SignUpPage';

it("should match snapshot with all data passed in correctly", () => {
  const page = shallow(<SignUpPage />)
  expect(page).toMatchSnapshot();
});
