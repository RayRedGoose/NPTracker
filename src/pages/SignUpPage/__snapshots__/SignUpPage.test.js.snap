// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with all data passed in correctly 1`] = `
<main
  className="signup-page"
>
  <Connect(SignUpBlock)
    id={1}
    inputs={
      Array [
        "username",
        "password",
      ]
    }
    key="1"
  />
  <Connect(SignUpBlock)
    id={2}
    inputs={
      Array [
        "name",
        "last_name",
        "city",
      ]
    }
    key="2"
  />
  <Connect(SignUpBlock)
    id={3}
    inputs={
      Array [
        "image",
      ]
    }
    key="3"
  />
  <Connect(SignUpBlock)
    id={4}
    inputs={Array []}
    key="4"
  />
</main>
`;
